src/reporter.cc

```diff
@@ -144,20 +144,32 @@ void MetadataReporter::SendMetadata(
 
   std::vector<json::value> entries;
   for (auto& entry : metadata) {
-    const MonitoredResource& resource = entry.first;
+    const std::string& full_resource_name = entry.first;
     MetadataStore::Metadata& metadata = entry.second;
     if (metadata.ignore) {
       continue;
     }
     json::value metadata_entry =
-        json::object({  // MonitoredResourceMetadata
-          {"resource", resource.ToJSON()},
-          {"rawContentVersion", json::string(metadata.version)},
-          {"rawContent", std::move(metadata.metadata)},
-          {"state", json::string(metadata.is_deleted ? "DELETED" : "ACTIVE")},
-          {"createTime", json::string(time::rfc3339::ToString(metadata.created_at))},
-          {"collectTime", json::string(time::rfc3339::ToString(metadata.collected_at))},
-        });
+        json::object({
+          {"name", json::string(full_resource_name)},
+          {"type", json::string(metadata.type)},
+          {"location", json::string(metadata.location)},
+          {"state", json::string(metadata.is_deleted ? "DELETED" : "EXISTS")},
+          {"createTime", json::string(
+              time::rfc3339::ToString(metadata.created_at))},
+          {"eventTime", json::string(
+              time::rfc3339::ToString(metadata.collected_at))},
+          {"views",
+            json::object({
+              {metadata.version,
+                json::object({
+                  {"schemaName", json::string(metadata.schema_name)},
+                  {"stringContent", json::string(metadata.metadata->ToString())},
```

Comment:
Using structContent would mean that the API is seeing a struct proto, not pure JSON. This would also mean conversion from JSON -> struct proto -> JSON, possibly leading to some data corruption. That is what I am trying to avoid here.

Reply:
Wouldn't the conversion be done at the ESF translation layer, before the API gets the data? I'm really wary of sending raw strings around — if this is what is required for fidelity, then this is bad API design.

Reply:
That is correct. I really want to prevent any conversion of the JSON to proto, since that can cause possible mismatches. Sending it as string means that the only conversion is back to JSON